ingestclient/core/engine.py

```diff
@@ -275,11 +283,25 @@ def run(self):
             raise Exception(msg)
 
         # Do some work
+        self.access_denied = False
+        self.access_denied_count = 0
+        self.invalid_access_key = False
+        self.invalid_access_key_count = 0
+
         wait_cnt = 0
         while True:
+            if self.access_denied:
+                self.access_denied = False
+                self.credential_create_time = datetime.datetime.min
+            if self.invalid_access_key:
+                self.invalid_access_key = False
+                if self.invalid_access_key_count % 5 == 4:
```

Comment:
Why get a new key for invalid access every 4 errors (vs every time for access denied)?

Reply:
I was assuming the invalid access key might have been happening from not waiting long enough for the keys to become valid.  I was hoping it may become valid by waiting longer.  Is that not how you read the error?

Reply:
May want to just put a comment to this effect, so it is clear in the future.